packages/forms-web-app/src/controllers/examination/check-submission-item/utils/get-back-link-url.js

```diff
@@ -0,0 +1,55 @@
+const {
+	getSubmissionItemType,
+	getSubmissionItemPersonalInformation,
+	getActiveSubmissionItemFiles,
+	getActiveSubmissionItem
+} = require('../../session/submission-items-session');
+const {
+	routesConfig: {
+		examination: {
+			directory,
+			pages: {
+				evidenceOrComment,
+				personalInformation,
+				personalInformationComment,
+				personalInformationCommentFiles,
+				personalInformationFiles,
+				personalInformationWhichCommentFiles,
+				personalInformationWhichFiles
+			}
+		}
+	}
+} = require('../../../../routes/config');
+
+const getBackLinkUrl = (session) => {
+	const activeSubmissionItem = getActiveSubmissionItem(session);
+	const submissionItemType = getSubmissionItemType(activeSubmissionItem);
+	const hasPersonalInformation =
+		getSubmissionItemPersonalInformation(activeSubmissionItem) ===
+		personalInformation.options[1].value;
+
+	let backLinkUrl;
+
+	if (submissionItemType === evidenceOrComment.options[1].value) {
+		backLinkUrl = `${directory}${personalInformationComment.route}`;
+	} else if (submissionItemType === evidenceOrComment.options[2].value) {
+		const submissionItemFilesLength = getActiveSubmissionItemFiles(session);
+
+		if (submissionItemFilesLength > 1 && hasPersonalInformation)
+			backLinkUrl = `${directory}${personalInformationWhichFiles.route}`;
+		else backLinkUrl = `${directory}${personalInformationFiles.route}`;
+	} else if (submissionItemType === evidenceOrComment.options[3].value) {
+		if (hasPersonalInformation)
+			backLinkUrl = `${directory}${personalInformationWhichCommentFiles.route}`;
+		else backLinkUrl = `${directory}${personalInformationCommentFiles.route}`;
+	}
+
+	if (!backLinkUrl)
+		throw new Error('Unable to assign back link URL as no required conditions are met');
```

Comment:
message: 'Unable to assign back link URL'

the rest i don't think helps and is implied.